### DISCUSSION TITLE
Agents over network

### DISCUSSION THREAD

**Discussion by mihaimyh** [microsoft/autogen#2767] [Ideas]
Can the communication between agents be done trough network? Meaning that, in a multiple agent setup, some agents can be deployed on different machines where they have to performs some tasks on that machine (ex: execute a custom script and report back the result).

**Comment by LittleLittleCloud** [comment]
Good question. In dotnet you can use Orleans to achieve that. Maybe in python there’s also some similar libraries that offer the same functionality?

Also the pr #2633 offers the ability of running an Open ai chat completion compatible service using an arbitrary agent . That solution should also work
